c_common/front_end_common_lib/include/spinn_extra.h

```diff
@@ -0,0 +1,1525 @@
+ASSERT_WORD_SIZED(router_dump_status_t);
+ASSERT_WORD_SIZED(router_diagnostic_counter_ctrl_t);
+ASSERT_WORD_SIZED(router_timing_counter_ctrl_t);
+ASSERT_WORD_SIZED(router_diversion_t);
+ASSERT_WORD_SIZED(router_fixed_route_routing_t);
+ASSERT_WORD_SIZED(router_diagnostic_filter_t);
+ASSERT_WORD_SIZED(router_multicast_route_t);
+ASSERT_WORD_SIZED(router_p2p_table_entry_t);
+
+//! Router controller registers
+static volatile router_t *const router_control = (router_t *) RTR_BASE;
+//! Router diagnostic filters
+static volatile router_diagnostic_filter_t *const router_diagnostic_filter =
+        (router_diagnostic_filter_t *) (RTR_BASE + 0x200);
+//! Router diagnostic counters
+static volatile uint *const router_diagnostic_counter =
+        (uint *) (RTR_BASE + 0x300);
+//! Router multicast route table
+static volatile router_multicast_route_t *const router_multicast_table =
+        (router_multicast_route_t *) RTR_MCRAM_BASE;
+//! Router multicast key table (write only!)
+static volatile uint *const router_key_table = (uint *) RTR_MCKEY_BASE;
+//! Router multicast mask table (write only!)
+static volatile uint *const router_mask_table = (uint *) RTR_MCMASK_BASE;
+//! Router peer-to-peer route table
+static volatile router_p2p_table_entry_t *const router_p2p_route_table =
+        (router_p2p_table_entry_t *) RTR_P2P_BASE;
+
+//! \}
+
+// ---------------------------------------------------------------------
+// 11. Inter-chip transmit and receive interfaces
+
+// No registers
+
+// ---------------------------------------------------------------------
+// 12. System NoC
+
+// No registers
+
+// ---------------------------------------------------------------------
+//! \name 13. SDRAM interface
+//! \brief The SDRAM interface connects the System NoC to an off-chip SDRAM
+//!     device. It is the ARM PL340, described in ARM document DDI 0331D.
+//! \details Only meaningfully usable by the monitor core when initialising the
+//!     overall chip. Use at other times is very much not recommended.
+//! \warning Do not use these without talking to Luis first!
+//! \{
+
+//! Memory controller status
+typedef struct {
+    uint status : 2;            //!< Config, ready, paused, low-power
+    uint width : 2;             //!< Width of external memory: 2’b01 = 32 bits
+    uint ddr : 3;               //!< DDR type: 3b’011 = Mobile DDR
+    uint chips : 2;             //!< Number of different chip selects (1, 2, 3, 4)
+    uint banks : 1;             //!< Fixed at 1’b01 = 4 banks on a chip
+    uint monitors : 2;          //!< Number of exclusive access monitors (0, 1, 2, 4)
+    uint : 20;                  // padding
+} sdram_status_t;
+
+//! Memory controller command
+typedef struct {
+    uint command : 3;           //!< one of ::sdram_command
+} sdram_command_t;
+
+//! \brief Memory controller commands, for ::sdram_command_t::command
+//! \todo Verify ::SDRAM_CTL_SLEEP, ::SDRAM_CTL_WAKE, ::SDRAM_CTL_ACTIVE_PAUSE
```

Comment:
did you do that? is it left for future? or is it a self reminder thats been left over?

Reply:
This is stuff that long-term we'll need to check with Luis. It's not pressing because current users of this include file are _not_ to touch the SDRAM controller! (It's done for completeness.)

Reply:
fair enough